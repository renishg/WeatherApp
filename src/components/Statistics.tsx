import React, {FC} from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {colors} from '../theme/colors';
import {fonts} from '../theme/fonts';
import {Card} from './Card';

type StatisticsProps = {
  data: {label: string; value: string}[];
};

const Statistics: FC<StatisticsProps> = ({data}) => {
  return (
    <Card styles={styles.container}>
      {data.map(({label, value}, index) => {
        const isLast = data.length === index + 1;
        return (
          <View
            key={index}
            style={StyleSheet.flatten([
              styles.infoContainer,
              !isLast ? styles.infoBorder : undefined,
            ])}>
            <Text style={styles.valueText}>{value}</Text>
            <Text style={styles.labelText}>{label}</Text>
          </View>
        );
      })}
    </Card>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  infoContainer: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  infoBorder: {
    borderRightColor: colors.white,
    borderRightWidth: 2,
  },
  valueText: {
    color: colors.white,
    fontSize: 20,
    fontFamily: fonts.bold,
  },
  labelText: {
    color: colors.white,
    fontSize: 16,
    fontFamily: fonts.regular,
  },
});

export {Statistics};
