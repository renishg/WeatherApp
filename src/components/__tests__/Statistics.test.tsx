import React from 'react';
import {View, Text} from 'react-native';
import {create} from 'react-test-renderer';
import {Statistics} from '../Statistics';

const mockData = [
  {label: 'Label 1', value: 'Value 1'},
  {label: 'Label 2', value: 'Value 2'},
  {label: 'Label 3', value: 'Value 3'},
];

describe('Statistics', () => {
  it('should render correctly', () => {
    const tree = create(<Statistics data={mockData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the correct number of data items', () => {
    const component = create(<Statistics data={mockData} />);
    const numberOfDataItems = component.root.findAllByType(View).length - 1;
    expect(numberOfDataItems).toEqual(mockData.length);
  });

  it('should render the correct label and value for each data item', () => {
    const component = create(<Statistics data={mockData} />);
    const infoContainers = component.root.findAllByType(View);
    infoContainers.shift();
    infoContainers.forEach((infoContainer, index) => {
      const AllTexts = infoContainer.findAllByType(Text);
      const valueText = AllTexts[0].props.children;
      const labelText = AllTexts[1].props.children;
      expect(valueText).toEqual(mockData[index].value);
      expect(labelText).toEqual(mockData[index].label);
    });
  });
});
