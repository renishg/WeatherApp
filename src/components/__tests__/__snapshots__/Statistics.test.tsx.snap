// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(0,0,0, 0.2)",
      "borderColor": "#FFFFFF",
      "borderRadius": 14,
      "borderWidth": 2,
      "flexDirection": "row",
      "marginVertical": 8,
      "paddingHorizontal": 12,
      "paddingVertical": 14,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderRightColor": "#FFFFFF",
        "borderRightWidth": 2,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans-Bold",
          "fontSize": 20,
        }
      }
    >
      Value 1
    </Text>
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
        }
      }
    >
      Label 1
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderRightColor": "#FFFFFF",
        "borderRightWidth": 2,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans-Bold",
          "fontSize": 20,
        }
      }
    >
      Value 2
    </Text>
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
        }
      }
    >
      Label 2
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans-Bold",
          "fontSize": 20,
        }
      }
    >
      Value 3
    </Text>
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
        }
      }
    >
      Label 3
    </Text>
  </View>
</View>
`;
